staging/src/k8s.io/kubectl/pkg/cmd/util/helpers.go

```diff
@@ -754,3 +760,27 @@ func Warning(cmdErr io.Writer, newGeneratorName, oldGeneratorName string) {
 		oldGeneratorName,
 	)
 }
+
+// ParseLabels turns a string representation of a label set into a map[string]string
+func ParseLabels(labelSpec interface{}) (map[string]string, error) {
```

Comment:
Maybe we could turn the function I've used in create_ingress (https://github.com/kubernetes/kubernetes/blob/v1.21.0/staging/src/k8s.io/kubectl/pkg/cmd/create/create_ingress.go#L287-L296) into something more global.

In this case, the function receive a string and convert into a map (needs more validation tho, I've just found a bug -> https://github.com/kubernetes/kubernetes/pull/101377)

This way we can always use the []string

Reply:
Or we can just use this function as I saw you've got it from the generators part :)

Anyway, I think using StringSlice is easier (and you cannot have ',' in labels anyway) but let me know what you think!